identity/authenticator.go

```diff
@@ -0,0 +1,33 @@
+package identity
+
+import (
+	"errors"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// Authenticator extracts identity
+type Authenticator interface {
+	Authenticate(message []byte, signature Signature) (Identity, error)
+}
+
+// NewAuthenticator extracts identity which was used to sign message
```

Comment:
This describes `Authenticator`, not `NewAuthenticator`.
*extracts identity which was used to sign message* - this is a useful information and could be moved to `Authenticator` or `Authenticate` method comment itself.

Reply:
Fixed